src/Tests/Microsoft.NET.Sdk.WorkloadManifestReader.Tests/SdkDirectoryWorkloadManifestProviderTests.cs

```diff
@@ -399,7 +422,105 @@ public void ItThrowsExceptionIfWorkloadSetJsonFilesHaveDuplicateManifests()
                 """);
 
             Assert.Throws<ArgumentException>(() =>
-                new SdkDirectoryWorkloadManifestProvider(sdkRootPath: _fakeDotnetRootDirectory, sdkVersion: "8.0.200", userProfileDir: null));
+                new SdkDirectoryWorkloadManifestProvider(sdkRootPath: _fakeDotnetRootDirectory, sdkVersion: "8.0.200", userProfileDir: null, globalJsonPath: null));
+        }
+
+        [Fact]
+        public void ItUsesWorkloadSetFromGlobalJson()
+        {
+            Initialize("8.0.200");
+
+            string? globalJsonPath = Path.Combine(_testDirectory, "global.json");
+            File.WriteAllText(globalJsonPath, """
+            {
+                "sdk": {
+                    "version": "8.0.200",
+                    "workloadversion": "8.0.201"
+                },
+                "msbuild-sdks": {
+                    "Microsoft.DotNet.Arcade.Sdk": "7.0.0-beta.23254.2",
+                }
+            }
+            """);
+
+            CreateMockManifest(_manifestRoot, "8.0.100", "ios", "11.0.1", true);
+            CreateMockManifest(_manifestRoot, "8.0.100", "ios", "11.0.2", true);
+            CreateMockManifest(_manifestRoot, "8.0.200", "ios", "12.0.1", true);
+
+            CreateMockWorkloadSet(_manifestRoot, "8.0.200", "8.0.201", """
+{
+  "ios": "11.0.2/8.0.100"
+}
+""");
+
+            CreateMockWorkloadSet(_manifestRoot, "8.0.200", "8.0.202", """
+{
+  "ios": "12.0.1/8.0.200"
+}
+""");
+
+            var sdkDirectoryWorkloadManifestProvider
+                = new SdkDirectoryWorkloadManifestProvider(sdkRootPath: _fakeDotnetRootDirectory, sdkVersion: "8.0.200", userProfileDir: null, globalJsonPath: globalJsonPath);
+
+            GetManifestContents(sdkDirectoryWorkloadManifestProvider)
+                .Should()
+                .BeEquivalentTo("ios: 11.0.2/8.0.100");
+        }
+
+        [Fact]
+        public void ItFailsIfWorkloadSetFromGlobalJsonIsNotInstalled()
+        {
+            Initialize("8.0.200");
+
+            string? globalJsonPath = Path.Combine(_testDirectory, "global.json");
+            File.WriteAllText(globalJsonPath, """
+            {
+                "sdk": {
+                    "version": "8.0.200",
+                    "workloadversion": "8.0.201"
```

Comment:
The global.json reader looks for `workloadVersion`. Shouldn't this through an exception because the property is all lower case?

Reply:
It's doing case-insensitive comparison.  I'm not sure whether global.json is supposed to be case-sensitive or not, but I followed the pattern from the workload manifest parsing code, which is case-insensitive.

Reply:
I went over the global.json docs. It doesn't mention anything from what I could tell. I'm fine if it's case-insensitive

Reply:
As I mentioned to you in chat, the runtime treats global.json properties as case sensitive. We should probably be consistent with the runtime.

Reply:
It's hard for me to see a downside for being case insensitive.  What would the would that break?  On the other hand, if we are case sensitive, some people will probably get the casing wrong and be confused as to why the setting isn't taking effect.

Reply:
We can mitigate that somewhat by updating the [Schemastore json schema for global.json](https://json.schemastore.org/global.json) with this field - this enables editors to flag this error.